jdaviz/app.py

```diff
@@ -973,6 +973,16 @@ def return_data_label(self, loaded_object, ext=None, alt_name=None, check_unique
 
         if check_unique:
             data_label = self.return_unique_name(data_label, ext=ext)
+            # It is possible to add data named "test (1)" and then
+            # add another data named "test" and return_unique_name will see the
+            # first test and assume the second is the duplicate, appending
+            # "(1)" to the end. This overwrites the original data and
+            # causes issues. This block appends a timestamp if a duplicate
+            # is still found in data_collection.
+            if data_label in self.data_collection.labels:
+                import time
+                ts = time.time()
+                data_label = f"{data_label} timestamp: {ts}"
```

Comment:
Thanks for the detailed comment which makes this fix easy to understand.

I'm wondering if this belongs in `return_data_label`. Would it be possible/preferable to put this catch in `return_unique_name`, so that `return_unique_name` returns a unique name.

Reply:
Ha, good point!